### PR TITLE
Fix test after Go 1.21 upgrade

### PR DESCRIPTION
**Pull request by jakule**
Updated the error messages in the lib/teleterm/teleterm_test.go and lib/auth/tls_test.go test files from "tls: bad certificate" to more specific ones ("tls: certificate required", "unknown certificate authority") to accurately reflect the Go 1.21 update in the TLS server behavior.

### CODE REVIEW

File: lib/auth/helpers.go [hunk 1]
```diff
@@ -718,6 +718,9 @@ func NewTestTLSServer(cfg TestTLSServerConfig) (*TestTLSServer, error) {
 	}
 	tlsConfig.Time = cfg.AuthServer.Clock().Now
 
+	// Go 1.21 changed the default behavior of TLS servers https://tip.golang.org/doc/go1.21
```
Comment (codingllama) [inline lib/auth/helpers.go:721]:
```suggestion
	// Go 1.21 changed the default behavior of TLS servers.
	// See https://go.dev/doc/go1.21#crypto/tls.
```

File: lib/auth/tls_test.go [hunk 2]
```diff
@@ -415,7 +415,7 @@ func TestAutoRotation(t *testing.T) {
 	// this is not going to be a problem in real teleport
 	// as it reloads the full server after reload
 	_, err = testSrv.CloneClient(proxy).GetNodes(ctx, apidefaults.Namespace)
-	require.ErrorContains(t, err, "bad certificate")
+	require.ErrorContains(t, err, "unknown certificate authority")
```
Comment (codingllama) [inline lib/auth/tls_test.go:418]:
```suggestion
	require.ErrorContains(t, err, "certificate")
```

File: lib/auth/tls_test.go [hunk 3]
```diff
@@ -588,7 +588,7 @@ func TestManualRotation(t *testing.T) {
...
```
Comment (codingllama) [inline lib/auth/tls_test.go:591]:
```suggestion
	require.ErrorContains(t, err, "certificate")
```

We don't care beyond that, or do we? Same for others.

File: lib/auth/tls_test.go [hunk 4]
```diff
@@ -683,7 +683,7 @@ func TestRollback(t *testing.T) {
 
 	// clients with new creds will no longer work
 	_, err = testSrv.CloneClient(newProxy).GetNodes(ctx, apidefaults.Namespace)
-	require.ErrorContains(t, err, "bad certificate")
+	require.ErrorContains(t, err, "unknown certificate authority")
```
Comment (codingllama) [inline lib/auth/tls_test.go:686]:
```suggestion
	require.ErrorContains(t, err, "certificate")
```

**Review by zmb3 — APPROVED** [review]

File: lib/teleterm/teleterm_test.go [hunk 5]
```diff
@@ -82,7 +82,7 @@ func TestStart(t *testing.T) {
 				return &tls.Config{InsecureSkipVerify: true}
 			},
 			connReadExpectationFunc: func(t *testing.T, connReadErr error) {
-				require.ErrorContains(t, connReadErr, "tls: bad certificate")
+				require.ErrorContains(t, connReadErr, "tls: certificate required")
```
Comment (codingllama) [inline lib/teleterm/teleterm_test.go:85]:
```suggestion
				require.ErrorContains(t, connReadErr, "tls:")
```

**Review by tigrato — APPROVED** [review]

File: lib/teleterm/teleterm_test.go [hunk 6]
```diff
@@ -93,7 +93,7 @@ func TestStart(t *testing.T) {
...
```
Comment (codingllama) [inline lib/teleterm/teleterm_test.go:96]:
```suggestion
				require.ErrorContains(t, connReadErr, "tls:")
```

**Review by codingllama — APPROVED** [review]
Thanks!

**Comment by codingllama** [comment]
Note: fixes breakages introduced by #30180.

**Pull request merged**